src/AddTaskDialog.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect } from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import TextField from '@mui/material/TextField';
+import Select from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import Box from '@mui/material/Box';
+import Rating from '@mui/material/Rating';
+import { v4 as uuidv4 } from 'uuid';
+
+
+const AddTaskDialog = ({ open, handleClose, onSave, editedTask }) => {
+
+    const [formData, setFormData] = useState({
```

Comment:
instead of useefect when you define the state check if it was opened for edit an exists task or to create a new one. by this info define your fields